File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/non-vul/code_before27.java
// From cwe-snippets, snippets_90/non-compliant/Java/cwe-0476/Integer_06.java

private void func() throws Throwable
    {
        Integer data;
        if (PRIVATE_STATIC_FINAL_FIVE==5)
        {
            data = null;
        }
        else
        {
            data = null;
        }

        if (PRIVATE_STATIC_FINAL_FIVE==5)
        {
            if (data != null)
            {
                IO.writeLine("" + data.toString());
            }
            else
            {
                IO.writeLine("data is null");
            }
        }
    }

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/c/non-vul/code_before27.c
// From cwe-snippets, snippets_90/non-compliant/C/cwe-0416/malloc_free_char_02.c

static void func()
{
    char * data;
    /* Initialize data */
    data = NULL;
    if(1)
    {
        data = (char *)malloc(100*sizeof(char));
        if (data == NULL) {exit(-1);}
        memset(data, 'A', 100-1);
        data[100-1] = '\0';
        free(data);
    }
    if(0)
    {
        printLine("Benign, fixed string");
    }
    else
    {
        /* do nothing */
    }
}